src/Controllers/Ops.php

```diff
@@ -101,40 +102,21 @@ protected function saveToFile($file, $data, $package_name = null)
         return true;
     }
 
-    protected function recSave($pre_format, $key, $code, $v)
+    private static function explodeTranlsationKey(string $nestedTranslationKey, string|null $value): array
     {
-        if (strpos($key, '.')) {
-            $assoc = [];
-            $exp   = explode('.', $key);
+        $doneSplitting = !Str::contains($nestedTranslationKey, '.');
 
-            // convert array dot to associative
-            while (!empty($exp)) {
-                $assoc = [array_pop($exp) => $assoc];
-            }
-
-            // set value to last key
-            Arr::set($assoc, $key, $v);
-
-            // get root key
-            $first_key = strtok($key, '.');
-
-            // hacky fix to avoid indexed nesting
-            $final = array_pop($assoc);
-            $a     = key($final);
-            $b     = current($final);
+        if ($doneSplitting) return [$nestedTranslationKey => $value];
 
-            // 3rd level deep
-            if (is_array($b)) {
-                $c = key($b);
-                $d = current($b);
+        $currentLevel = collect(explode('#', Str::replaceFirst('.', '#', $nestedTranslationKey)));
+        $collection = collect([$currentLevel[0] => self::explodeTranlsationKey($currentLevel[1], $value)]);
```

Comment:
we build an array with the most left part (before the first dot/hash) as the key and the return value of this same function as value.

For the key 'abc.fgh.lmn.pqr' this line is reached 3 times once with the currentlevel 'abc' once with 'fgh' and once with 'lmn'